Neo4j.Driver/Neo4j.Driver/Internal/AsyncSession.cs

```diff
@@ -302,6 +311,12 @@ private async Task<Bookmarks> GetBookmarksAsync()
         AccessMode mode,
         Func<IAsyncTransaction, Task<T>> work,
         Action<TransactionConfigBuilder> action)
+    {
+        using var _ = Driver.TelemetryManager.StartApiActivity(QueryApiType.TransactionFunction);
+        return RunTransactionImplAsync(mode, work, action);

```

Comment:
if RunTransactionImplAsync awaits before reading the activity it will be unset so best to await here